zerver/webhooks/thinkst/view.py

```diff
@@ -0,0 +1,52 @@
+# Webhooks for external integrations.
+from typing import Any, Dict, Optional
+
+from django.http import HttpRequest, HttpResponse
+
+from zerver.decorator import api_key_only_webhook_view
+from zerver.lib.request import REQ, has_request_variables
+from zerver.lib.response import json_success
+from zerver.lib.webhooks.common import check_send_webhook_message
+from zerver.models import UserProfile
+
+
+@api_key_only_webhook_view('Thinkst')
+@has_request_variables
+def api_thinkst_webhook(request: HttpRequest, user_profile: UserProfile,
+                        message: Dict[str, Any]=REQ(argument_type='body'),
+                        user_specified_topic: Optional[str]=REQ('topic', default=None)) -> HttpResponse:
+
+    body = None
+    topic = None
+
+    # There are two types of requests that Thinkst will send - depending on whether it is
+    # a canary or a canarytoken. Unfortunately, there isn't a great way to differentiate other
+    # than look at the contents.
+    is_canarytoken = 'Timestamp' not in message
```

Comment:
This could be extracted to a function of it's own. It's not necessary, but I think it would be nice modularization of the logic.

Reply:
Fixed, thanks!